modules/custom/az_event/az_event.module

```diff
@@ -95,6 +95,67 @@ function az_event_node_view(array &$build, EntityInterface $entity, EntityViewDi
         ];
       }
     }
+
+    // For the full content view mode, modify the display of recurring events.
+    if (($view_mode === 'full') && ($entity instanceof NodeInterface)) {
+      /** @var \Drupal\Core\Field\FieldItemList $eventDates */
+      $eventDates = $entity->field_az_event_date;
+      $numberOfRecurringInstances = $eventDates->count();
+      if ($numberOfRecurringInstances > 1) {
+        // Initialize the accordion body build array.
+        $accordionBody = $build['field_az_event_date'];
+        foreach ($entity->field_az_event_date as $item) {
+          unset($accordionBody[$item->getName()]);
+        }
+        $itemIndex = 0;
+
+        // Add any upcoming instances to the accordion body.
+        $nextRecurringInstance = NULL;
+        $now = \Drupal::time()->getCurrentTime();
+        foreach ($entity->field_az_event_date as $item) {
+          if ($item->end_value > $now) {
+            if (!isset($nextRecurringInstance)) {
+              $nextRecurringInstance = $item->getName();
+              $accordionBody[$itemIndex++] = [
+                '#markup' => '<h3 class="h5 mt-0">Upcoming Events</h3>',
+              ];
+            }
+            $accordionBody[$itemIndex++] = $build['field_az_event_date'][$item->getName()];
+          }
+        }
+
+        // Add any past instances to the accordion body.
+        if (!isset($nextRecurringInstance) || $nextRecurringInstance !== 0) {
+          $accordionBody[$itemIndex++] = [
+            '#markup' => "<h3 class='h5" . (isset($nextRecurringInstance) ? '' : ' mt-0') . "'>Past Events</h3>",
```

Comment:
Should these strings be passed to `TranslatableMarkup` so they can be localized?